### DISCUSSION TITLE
Copier 6 first beta out today! Testers needed

### DISCUSSION THREAD

**Discussion by yajo** [copier-org/copier#640] [Show and tell]
Hello community! I know many of you have been using copier 6 alpha releases for a long time. Actually that's quite amazing for me, as the current maintainer. It means you really love Copier ❤. Or that you really needed those shiny new features! Or both...

V6 has been [the biggest milestone](https://github.com/copier-org/copier/issues?q=milestone%3Av6.0.0) for the whole project, and it's almost ended!

Today I released [the 1st beta](https://github.com/copier-org/copier/releases/tag/v6.0.0b0). From now on, I won't accept new features until v6 final is out (got some? open it, 😉 it'll just be deferred).

![imagen](https://user-images.githubusercontent.com/973709/163326095-9491117d-c58f-4dd6-be4b-e078f01be643.png)

I want to ask for **beta testers**. I expect Copier 6 to have bugs, so it doesn't have to be perfect to be released. But it breaks some things, so I think it's good to let users some time before release.

What can you do if you want to help?

- Tell your friends.
- Test your v5 templates on V6.
- If something goes wrong, go to [the latest docs](https://copier.readthedocs.io/en/latest/) and try to solve it with that help.
- Incomplete docs? Open issue or PR please.
- Found a bug? Do the same please.
- Did it go well? Or did you manage to solve the problem? Drop a comment around here telling your experience and feelings about the update.
- Are you in love with Copier? Make adonation. 😍

After some time and feedback, when I consider everything reasonably stable, I'll publish the next stable! 🎉

Thanks!

**Comment by legalsylvain** [comment]
whaou ! huge work ! thanks !

**Comment by yeedle** [comment]
Works flawlessly! thanks!

**Comment by nwithers-ecr** [comment]
Running into a problem where copier is infinitely recusing and creating directories. I am trying to test my template locally, so I'm running copier like so `copier . .`. this works as expected. However I would prefer for the template to go into an existing directory. So I created a directory called `test` and ran `copier . ./test` and got this output.
```
    create  test
    create  test/test
    create  test/test/test
    create  test/test/test/test
    create  test/test/test/test/test
    create  test/test/test/test/test/test
    create  test/test/test/test/test/test/test
    create  test/test/test/test/test/test/test/test
    create  test/test/test/test/test/test/test/test/test
    create  test/test/test/test/test/test/test/test/test/test
    create  test/test/test/test/test/test/test/test/test/test/test
    create  test/test/test/test/test/test/test/test/test/test/test/test
```
this will go forever until I imagine it hits the Linux filename limit, though I haven't let it run that long.
If I try to name it something different, `copier . ./test/nope` it recurses like so
```
    create  .
    create  test
    create  test/nope
    create  test/nope/test
    create  test/nope/test/nope
    create  test/nope/test/nope/test
    create  test/nope/test/nope/test/nope
    create  test/nope/test/nope/test/nope/test
    create  test/nope/test/nope/test/nope/test/nope
```
This is only the case when the copier template is a parent of the destination. Doing something like this `copier . ../test` works as expected.

With 5.1.0 i didn't see this recursive problem.

**Reply by pawamoy** [reply]
Did you try to add the test directory to .gitignore? I think it will prevent Copier from recursing on it.

**Reply by nwithers-ecr** [reply]
Unfortunately, using a .gitignore of
```
test/
test/**
```
did not work.
I think I may have found the source of the problem though. I have my git repo checked out as bare and am running copier inside a worktree. I tried cloning the repo as normal running `copier . test` in that instance seemed to work. 
I noticed plumbum is using a --git-dir assuming it is located in the current directory. For bare repos this is not the case, and actually resides at the filepath inside the `.git` file. 

(this error isn't related to the recursion)
```
File "/home/nwithers/.local/pipx/venvs/copier/lib/python3.8/site-packages/plumbum/machines/base.py", line 28, in verify
    raise ProcessExecutionError(
plumbum.commands.processes.ProcessExecutionError: Unexpected exit code: 128
Command line: | /usr/bin/git --git-dir=.git --work-tree=/path_to_worktree/test commit -m 'Copier automated commit for draft changes' --no-verify
```
~I have no idea what plumbum does or if its at all related, just something I observed~

**Reply by nwithers-ecr** [reply]
looks like it might be this line https://github.com/copier-org/copier/blob/v6.0.0/copier/vcs.py#L142 and L144
I can try to build copier locally and see if following the `.git` file fixes it

**Comment by yajo** [comment]
Try adding that directory to be excluded by copier. See https://copier.readthedocs.io/en/latest/configuring/#exclude

**Reply by nwithers-ecr** [reply]
This seems to work, thank you.
